lib/Sema/CSFix.cpp

```diff
@@ -759,6 +759,24 @@ IgnoreContextualType *IgnoreContextualType::create(ConstraintSystem &cs,
       IgnoreContextualType(cs, resultTy, specifiedTy, locator);
 }
 
+bool RemoveUnnecessaryCoercion::diagnose(Expr *root, bool asNote) const {
+  auto &cs = getConstraintSystem();
+
+  UnnecessaryCoercionFailure failure(root, cs,
```

Comment:
This code doesn't seem properly aligned... Could you please use clang-format (or even git-clang-format which is part of llvm tools) to adhere to our established code formatting?

Reply:
Also you can use `ContextualMismatch` and `ContextualFailure` as parent types of both fix and diagnostic they already have storage for "from" and "to" types.

Reply:
Done.